packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +76,29 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+export function scrollIntoViewport(targetElement: Element, containingElement?: Element) {
+  // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+  if (!isScrollPrevented) {
+    let {left: originalLeft, top: originalTop} = targetElement.getBoundingClientRect();
+
+    // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()
+    // won't cause a scroll if the element is already focused and doesn't behave consistently when an element is partially out of view horizontally vs vertically
+    targetElement?.scrollIntoView?.({block: 'nearest'});
+    let {left: newLeft, top: newTop} = targetElement.getBoundingClientRect();
+    // Account for sub pixel differences from rounding
+    if ((Math.abs(originalLeft - newLeft) > 1) || (Math.abs(originalTop - newTop) > 1)) {
```

Comment:
Note that the original positioning and the new positioning of the target element won't differ at all if the target was "out of view" from pinch zooming. It will still be scrolled into view due to `targetElement?.scrollIntoView?.({block: 'nearest'});` but the position w/ respect to the layout viewport is still the same.